modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/configuration/ConfigurationManagerBean.java

```diff
@@ -237,6 +238,7 @@ public void completePluginConfigurationUpdate(PluginConfigurationUpdate update)
             // Flush before merging to ensure the update has been persisted and avoid StaleStateExceptions.
             entityManager.flush();
             entityManager.merge(update);
+            entityManager.remove(oldPluginConfig);
```

Comment:
Wouldn't we want the old plug-in configuration to remain in the plug-in config history page of the resource? It seems that a configuration update/merge would not result in an orphaned configuration object. Granted, if the configuration object has no associated configuration parameters, I would consider the update to be strange/broken.